core/server/models/base/index.js

```diff
@@ -494,6 +495,13 @@ ghostBookshelf.Model = ghostBookshelf.Model.extend({
             slug = (slug.indexOf('-') > -1) ? slug.substr(0, slug.indexOf('-')) : slug;
         }
 
+        // TODO: remove the labs requirement when internal tags is out of beta
+        // This checks if the first character of a tag name is a #. If it is, this
+        // is an internal tag, and as such we should add 'hash' to the beginning of the slug
+        if (labs.isSet('internalTags') && baseName === 'tag' && /^#/.test(base)) {
```

Comment:
Sorry.. I just thought of this. We should only do this if `options.importing` is not set, the same as [here](https://github.com/acburdine/Ghost/blob/ab9d0b646485813faee766296a675f6e933d32d8/core/server/utils/index.js#L72).